openwrt-openwisp-monitoring/Makefile

```diff
@@ -0,0 +1,76 @@
+# openwisp.org
+#
+# This is free software, licensed under the GNU General Public License v3.
+# See /LICENSE for more information.
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=openwrt-openwisp-monitoring
+PKG_VERSION:=$(shell cat ../VERSION)
+PKG_RELEASE:=1
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)/$(PKG_SOURCE_SUBDIR)
+PKG_MAINTAINER:=Federico Capoano <f.capoano@openwisp.io>
+PKG_LICENSE:=GPL-3.0-or-later
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/openwisp-monitoring
+	TITLE:=OpenWISP Monitoring
+	CATEGORY:=Administration
+	SECTION:=admin
+	SUBMENU:=openwisp
+	PKGARCH:=all
+	URL:=http://openwisp.org
+endef
```

Comment:
I don't see dependencies defined for this package.
Isn't it described in the project idea that this package should depend on openwisp-config?

You see, without defining this dependency, if you try to install only openwisp-monitoring, nothing will work, because curl will not be installed (unless you have installed it manually). 

OpenWISP Config defines a set of dependencies: https://github.com/openwisp/openwisp-config/blob/master/openwisp-config/Makefile#L21

We also rely on the openwisp-config configuration to get the URL, UUID and key.

If you look at the structure of the Makefile of openwisp-config, you will see there are 4 packages, one for each SSL library supported by openwisp and one which does not depend on any SSL library and can used only for plain HTTP (no SSL).

This means that we'll have to replicate the same here, we'll need:

- openwisp-monitoring-openssl
- openwisp-monitoring-mbedtls
- openwisp-monitoring-wolfssl
- openwisp-monitoring-nossl

This is very common in OpenWRT, look also at openvpn or other packages who need SSL support.
You may ask yourself why, the reason is that openssl is huge compared to `wolfssl` and `mbedtls`, which are lighter and less fully featured implementations which are suited for embedded development where in many cases available space is limited.